### DISCUSSION TITLE
How do I get current user ID?

### DISCUSSION THREAD

**Discussion by amr3k** [fastapi-users/fastapi-users#522] [Q&A] [answered]
Hi..
I'm following documentation and using cookie authentication ..
How do I get current user ID so I can return some data related to that user?

**Comment by amr3k** [comment, marked as answer]
Found the solution by looking at `fastapi_users/router/users.py` in the `get_users_router()` function, specifically in this part:
```
@router.get("/me", response_model=user_model)
    async def me(
        user: user_db_model = Depends(get_current_active_user),  # type: ignore
    ):
        return user
```
I'm still learning so I really don't know why this works?

**Reply by frankie567** [reply]
All the details about the usage details of the dependency are here: https://frankie567.github.io/fastapi-users/usage/dependency-callables/

As of how it works, well, it's the magic of FastAPI Users 🧙‍♂️ Schematically, the dependency works like this:

* It goes through all your authentication backends and looks for the corresponding authentication info (in a cookie, in a header...)
* If it finds one, it decodes it. Basically, it's a JWT containing the user id.
  * Otherwise, it raises a 401 error (unless you specified `optional`)
* Then, it looks for this user in the database.
* If the user is found, it returns it.
